### DISCUSSION TITLE
Allowed modifications on the code

### DISCUSSION THREAD

**Discussion by danymat** [Syslifters/sysreptor#260] [Q&A]
Hello Syslifters :)

While i'm trying to get Sysreptor Pro in the team (btw, I guess we would be booking a demo soon), we are already using community mode and pushing the need of the Pro by showing the usability, the effectiveness and the benefits regarding our previous generation tools.
In that regard, what is allowed to change/add in the frontend (on community/pro) ?
My first ideas are to:
- Change sysreptor logo to internal one (gitlab and other tools allow this)
- Change theme colors to comply with branding colors
- Add tabs to the left of the application (containing internal code, for internal processes)

I've read the license and the part related to my request is:

https://github.com/Syslifters/sysreptor/blob/6097e6e78d3db7d77efe75d477d9999e32971e0a/LICENSE#L8C1-L17

**Comment by aronmolnar** [comment]
Hi Danymat,

Thanks for your input and your ideas.  

We regard it as acceptable use for Professional licenses to change the logo and the colors, but we unfortunately don't allow changing any Python code.

We will be, however, working on a plugin system for the SysReptor server which will allow you exactly this (add custom logic, menu entries, etc.; we might schedule this for H2/24). In the meantime you can build your own logic in the reptor CLI (https://github.com/syslifters/reptor/), which you can use to build custom plugins.

For customization of logo and CSS, the following file must be edited:
* https://github.com/Syslifters/sysreptor/tree/main/frontend/src/public/static (favicon.svg and logo.svg)
* https://github.com/Syslifters/sysreptor/tree/main/frontend/src/components/Svg (Logo.vue and LogoText.vue)
* https://github.com/Syslifters/sysreptor/blob/main/frontend/src/vuetify.config.ts

Note that this requires you to recompile the application which will increase your updating efforts. Additionally, we cannot provide any warranty for customizations and don't recommend them.

It might also be feasible to inject custom CSS styles via a reverse proxy, maybe also the logo, without changing the source code.

I hope this helps.
